lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,213 @@
+ * under the License.
+ */
+/*
+ * Imports.
+ */
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import * as React from "react";
+import { PureComponent, Fragment } from "react";
+import SignInUpTheme from "../../themes/signInUp";
+import FeatureWrapper from "../../../../../components/featureWrapper";
+import { StyleProvider } from "../../../../../styles/styleContext";
+import { defaultPalette } from "../../../../../styles/styles";
+import { getQueryParams, getRedirectToPathFromURL } from "../../../../../utils";
+import Recipe from "../../../recipe";
+import { getStyles } from "../../../components/themes/styles";
+import { RecipeInterface, LoginAttemptInfo } from "../../../types";
+import { ComponentOverrideContext } from "../../../../../components/componentOverride/componentOverrideContext";
+import { FeatureBaseProps } from "../../../../../types";
+import { formatPhoneNumberIntl } from "react-phone-number-input/min";
+
+type SignInUpState = {
+    error: string | undefined;
+    loaded: boolean;
+    loginAttemptInfo: LoginAttemptInfo | undefined;
+};
+
+type PropType = FeatureBaseProps & {
+    recipe: Recipe;
+};
+
+class SignInUp extends PureComponent<PropType, SignInUpState> {
+    constructor(props: PropType) {
+        super(props);
+
+        let error: string | undefined = undefined;
+        const errorQueryParam = getQueryParams("error");
+        if (errorQueryParam !== null) {
+            if (errorQueryParam === "signin") {
+                error = "Something went wrong. Please try again";
+            } else if (errorQueryParam === "restart") {
+                error = "The link you tried to click expired or was revoked. Please try again.";
+            }
+        }
+        this.state = {
+            loaded: false,
+            loginAttemptInfo: undefined,
+            error,
+        };
+    }
+
+    getIsEmbedded = (): boolean => {
+        if (this.props.isEmbedded !== undefined) {
+            return this.props.isEmbedded;
+        }
+        return false;
+    };
+
+    getModifiedRecipeImplementation = (): RecipeInterface => {
+        return {
+            ...this.props.recipe.recipeImpl,
+            createCode: async (input) => {
+                let contactInfo;
+                if ("email" in input) {
+                    contactInfo = input.email;
+                } else {
+                    contactInfo = formatPhoneNumberIntl(input.phoneNumber);
+                }
+                const res = await this.props.recipe.recipeImpl.createCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...res,
+                        lastResend: new Date().getTime(),
+                        contactInfoType: this.props.recipe.config.contactInfoType,
+                        contactInfo,
+                    };
+                    this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
```

Comment:
Need to put await

Reply:
This kind of error is too easy to make, I think we should a [this rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md) to the linter. I'll do this later because it reports this in a few other places (where they are hopefully intentional)